underfs/s3/src/test/java/tachyon/underfs/s3/S3UnderFileSystemFactoryTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.underfs.s3;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import tachyon.conf.TachyonConf;
+import tachyon.underfs.UnderFileSystemFactory;
+import tachyon.underfs.UnderFileSystemRegistry;
+
+/**
+ * This test ensures the s3 ufs module correctly accepts paths that begin with s3n://
+ */
+public class S3UnderFileSystemFactoryTest {
+  @Test
+  public void factoryTest() {
+    TachyonConf conf = new TachyonConf();
+
+    UnderFileSystemFactory factory =
+        UnderFileSystemRegistry.find("s3n://localhost/test/path", conf);
+    Assert.assertNotNull(
+        "A UnderFileSystemFactory should exist for s3n paths when using this module", factory);
+  }
```

Comment:
I'm usually against negative tests because there's technically the space of all possible strings, but perhaps one negative case is sufficient to ensure that the factory isn't just returning true for everything. What do you think?